django/core/mail/message.py

```diff
@@ -100,22 +100,62 @@ def forbid_multi_line_headers(name, val, encoding):
     return str(name), val
 
 
+def split_addr(addr, encoding):
+    """
+    Splits an address into local part and domain, properly encoded.
+    """
+    if '@' in addr:
+        localpart, domain = addr.split('@', 1)
+        localpart = Header(localpart, encoding).encode()
+        domain = domain.encode('idna').decode('ascii')
+    else:
+        localpart = Header(addr, encoding).encode()
+        domain = ''
+    return (localpart, domain)
+
+
 def sanitize_address(addr, encoding):
+    """
+    Formats a pair of (name, address) or an e-mail address string.
+
+    On Python 3, an `email.headerregistry.Address` object is used since
```

Comment:
I would move these to inline comments near the imports as they can probably be mostly removed when dropping Python 2 and simplifying the code.